src/installer/image/installer.go

```diff
@@ -17,35 +19,45 @@ import (
 
 type Properties struct {
 	ImageUri     string
+	ImageDigest  string
 	DockerConfig dockerconfig.DockerConfig
+	PathResolver metadata.PathResolver
 }
 
-func NewImageInstaller(fs afero.Fs, props *Properties) *imageInstaller {
-	return &imageInstaller{
+func NewImageInstaller(fs afero.Fs, props *Properties) *ImageInstaller {
+	return &ImageInstaller{
 		fs:    fs,
 		props: props,
 	}
 }
 
-type imageInstaller struct {
+type ImageInstaller struct {
 	fs    afero.Fs
 	props *Properties
 }
 
-func (installer *imageInstaller) InstallAgent(targetDir string) error {
-	log.Info("installing agent", "target dir", targetDir)
-	if err := installer.installAgentFromImage(targetDir); err != nil {
+func (installer *ImageInstaller) ImageDigest() string {
+	return installer.props.ImageDigest
+}
+
+func (installer *ImageInstaller) InstallAgent(targetDir string) error {
```

Comment:
```suggestion
func (installer ImageInstaller) InstallAgent(targetDir string) error {
```

Reply:
it modifies the state (its not nice but must happen) 😞